### PR TITLE
Add poetry requirements to build-system requirements

### PR DESCRIPTION
**Pull request by thejcannon**
This is a "fix" for #2072 , however I don't really consider it a long-term fix. I have verified it does allow `pip install`-ing the package however, by doing just that t this fork 😄.

# Pull Request Check List

This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://python-poetry.org/docs/contributing/) at least once, it will save you unnecessary review cycles!

- [ ] ~Added **tests** for changed code.~
- [ ] ~Updated **documentation** for changed code.~

**Note**: If your Pull Request introduces a new feature or changes the current behavior, it should be based
on the `develop` branch. If it's a bug fix or only a documentation update, it should be based on the `master` branch.

If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -77,7 +77,21 @@ poetry = "poetry.console:main"
 
 
 [build-system]
-requires = ["intreehooks"]
+requires = [
+    "intreehooks",
+    # According to PEP 517 we shouldn't rely on external dependencies.
```
Comment (thejcannon) [inline pyproject.toml]: Should a comment be made above where we list the requirements to also update this list?

**Comment by thejcannon** [comment]
If we _really_ wanted to, a test could be written that uses [`modulefinder`](https://docs.python.org/3.6/library/modulefinder.html) to find what packages `poetry/masonry/api.py` uses, and ensure this list matches the result. Additionally, the test(s) could correlate the `dependencies` caret versions with the PEP 440 versions here.

It would programmatically ensure this package could be `pip` installed from Git. But is also hardcore.

**Comment by adisbladis** [comment]
This "fix" will make it almost impossible to properly solve environments where the _application_ is depending on any of these libraries.

My two cents: The vendoring approach used by `get-poetry.py` (and the Nix expressions I maintain to package Poetry) seems to be the only sane way to separate _application dependencies_ from _package manager dependencies_.

**Comment by thejcannon** [comment]
If I understand correctly, that would only be true if the user has installed `poetry` in the same environment as their application, right? In that case, they already have these dependencies.

Isn't the advice to have `poetry` installed outside of the application environment?

**Comment by finswimmer** [comment]
Hello @thejcannon,

in the meantime the issue you've linked is fixed. So I'm closing yours.

Thanks a lot for your contribution :+1: 

fin swimmer

**Comment by thejcannon** [comment]
Thanks a lot for the tool 👍

**Pull request closed without merging**